uTinyRipperCore/Parser/Classes/TrailRenderer/TrailRendererParamaters.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using uTinyRipper.AssetExporters;
+using uTinyRipper.Classes;
+using uTinyRipper.Classes.AnimationClips;
+using uTinyRipper.Classes.ParticleSystems;
+using uTinyRipper.SerializedFiles;
+using uTinyRipper.YAML;
+using Object = uTinyRipper.Classes.Object;
+
+namespace uTinyRipper.Parser.Classes.TrailRenderers
+{
+	public struct TrailRendererParameters : IAssetReadable, IYAMLExportable
+	{
+		public static int GetSerializedVersion(Version version)
+		{
+			if (version.IsGreaterEqual(2018, 3))
+			{
+				return 3;
+			}
+			if (version.IsGreaterEqual(5))
+			{
+				return 2;
+			}
+			return 1;
+		}
+		/// <summary>
+		/// 2019.2.0 and greater
+		/// </summary>
+		public static bool IsReadShadowBias(Version version)
+		{
+			return version.IsGreaterEqual(2018, 3);
+		}
+		public void Read(AssetReader reader)
+		{
+			WidthMultiplier = reader.ReadSingle();
+			WidthCurve.Read(reader);
+			ColorGradient.Read(reader);
+			NumCornerVertices = reader.ReadInt32();
+			NumCapVertices = reader.ReadInt32();
+			Alignment = (LineAlignment)reader.ReadInt32();
+			TextureMode = (LineTextureMode)reader.ReadInt32();
+			if (IsReadShadowBias(reader.Version))
+			{
+				ShadowBias = reader.ReadSingle();
+			}
+			GenerateLightingData = reader.ReadBoolean();
+			reader.AlignStream(AlignType.Align4);
+		}
+		public YAMLNode ExportYAML(IExportContainer container)
+		{
+			YAMLMappingNode node = new YAMLMappingNode();
+			node.AddSerializedVersion(GetSerializedVersion(container.ExportVersion));
+			node.Add(WidthMultiplierName, WidthMultiplier);
+			node.Add(WidthCurveName, WidthCurve.ExportYAML(container));
+			node.Add(ColorGradientName, ColorGradient.ExportYAML(container));
+			node.Add(NumCornerVerticesName, NumCornerVertices);
+			node.Add(NumCapVerticesName, NumCapVertices);
+			node.Add(AlignmentName, (int)Alignment);
+			node.Add(TextureModeName, (int)TextureMode);
+			if (IsReadShadowBias(container.Version))
+			{
+				node.Add(ShadowBiasName, ShadowBias);
```

Comment:
SerializedVersion was changed when ShadowBias field was added. I think this field should has other value for previous versions.

Reply:
What do you mean?

Reply:
i mean that unitech sometime changes default value or value interpretation of the fields. imagine there is a float filed that has value range from 0 to 1. and unitech decided to change its range to from 0 to 100. to do it, they increase serialize version number. so,  during import process, new version of unity editor detect old serialized version, read value and multiply it to 100, to make it fit new range.
since serialized version of trail renderer parameter had been changed, some field value should be converted to new format.

Reply:
I'm not sure how to find that out, so I haven't added a fix for that in the latest commit